ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -294,16 +296,7 @@ let pool_migrate ~__context ~vm ~host ~options =
             end;
           with _ -> ()
         end;
-        match exn with
-        | Xenops_interface.Xenopsd_error Failed_to_acknowledge_shutdown_request ->
-          raise Api_errors.(Server_error (vm_failed_shutdown_ack, [Ref.string_of vm]))
-        | Xenops_interface.Xenopsd_error Cancelled _ ->
-          TaskHelper.raise_cancelled ~__context
-        | Xenops_interface.Xenopsd_error Storage_backend_error (code, _) ->
-          raise Api_errors.(Server_error (sr_backend_failure, [code]))
```

Comment:
I checked the other exceptions and they are handled similarly by `transform_xenops_exn` (IIUC then Canceled is handled better because it looks up the id), but the storage exception is handled differently.
`transform_xenops_exn` would reraise using the `code`, and this code would reraise as `sr_backend_failure`. Should we update `transform_xenopsd_exn`?

Reply:
Hmm, indeed. @jonludlam, is SR_BACKEND_FAILURE the API error that we need for all cases `Xenopsd_error Storage_backend_error`?

Reply:
I think actually `Storage_backend_error` is a wrapper for real `Api_error` errors, so the new code after this PR is doing it correctly.